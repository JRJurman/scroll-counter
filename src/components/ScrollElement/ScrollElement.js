import { registerHtml } from 'tram-one'
import './ScrollElement.scss'
const html = registerHtml()

export default ({ counterId, value }) => {
  return html`
    <div class="ScrollElement" counter-id=${counterId} value=${value}>
      ${value}
    </div>
  `
}
