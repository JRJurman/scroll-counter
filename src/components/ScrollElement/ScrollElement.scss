.ScrollElement {
  width: 42%;
  font-size: 2em;
  flex-shrink: 0;
  color: inherit;
  text-align: center;
  scroll-snap-align: center;
}
