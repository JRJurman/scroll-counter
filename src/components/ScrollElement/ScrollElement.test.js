import ScrollElement from './ScrollElement'

describe('ScrollElement', () => {
  describe('with default props', () => {
    const wrapper = ScrollElement({ counterId: 0, value: 20 })

    it('should match snapshot', () => {
      expect(wrapper.outerHTML).toMatchSnapshot()
    })
  })
})
