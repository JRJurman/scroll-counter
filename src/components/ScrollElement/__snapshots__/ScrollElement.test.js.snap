// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollElement with default props should match snapshot 1`] = `"<div counter-id=\\"0\\" value=\\"20\\" class=\\"ScrollElement\\">20</div>"`;
